docs/src/imagesegmentation.md

```diff
@@ -0,0 +1,317 @@
+julia> img = testimage("lena_gray")
+julia> function homogeneous(img)
+           min, max = extrema(img)
+           max - min < 0.2
+       end
+julia> seg = region_splitting(img, homogeneous);
+```
+
+**Original:**
+
+![Original](assets/segmentation/lena.jpg)
+
+**Segmented Image with labels replaced by their intensity means:**
+
+![SegmentedImage](assets/segmentation/lena_seg.jpg)
+
+#### Fuzzy C-means
+
+Fuzzy C-means clustering is widely used for unsupervised image segmentation. It is an
+iterative algorithm which tries to minimize the cost function:
+
+```math
+J = \displaystyle\sum_{i=1}^{N} \sum_{j=1}^{C} \delta_{ij} \| x_{i} - c_{j} \|^2
+```
+
+Unlike K-means, it allows pixels to belong to two or more clusters. It is widely used
+for medical imaging like in the soft segmentation of brain tissue model.
+
+**Time Complexity:** ``O(n*C^2*iter)`` where ``n`` is the number of pixels, ``C`` is
+number of clusters and ``iter`` is the number of iterations.
+
+###### Demo
+
+```julia
+julia> using ImageSegmentation, Images;
+julia> img = load("ring.jpg");
+julia> r = fuzzy_cmeans(img, 4, 1.8);
+```
+
+**Original**
+
+![Original](assets/segmentation/ring.jpg)
+
+**Output with pixel intensity = cluster center intensity * membership of pixel in that class**
+
+![SegmentedImage1](assets/segmentation/ring_s1.jpg) ![SegmentedImage2](assets/segmentation/ring_s3.jpg)
+![SegmentedImage3](assets/segmentation/ring_s4.jpg) ![SegmentedImage4](assets/segmentation/ring_s2.jpg)
+
+#### Watershed
+
+The watershed algorithm treats an image as a topographic surface where bright pixels correspond to peaks and dark pixels correspond to valleys. The algorithm starts flooding from valleys (local minima) of this topographic surface and region boundaries are formed when water from different sources merge. If the image is noisy, this approach leads to oversegmetation. To prevent oversegmentation, marker-based watershed is used i.e. the topographic surface is flooded from a predefined set of markers.  
+
+###### Demo
+
+```julia
+using Images, ImageSegmentation;
+
+img = load(download("http://docs.opencv.org/3.1.0/water_coins.jpg"));
+bw = Gray.(img).>0.5;
+dist = 1.-distance_transform(feature_transform(bw));
+markers = label_components(dist.<-15);
+segments = watershed(dist, markers);
+```
+
+![img1](assets/segmentation/water_coins.jpg) ![img2](assets/segmentation/watershed.jpg)
+
+## Some helpful functions
+
+#### Creating a Region Adjacency Graph (RAG)
+
+A region adjacency graph can directly be constructed from a `SegmentedImage`
+using the `region_adjacency_graph` function. Each segment is denoted by a vertex
+and edges are constructed between adjacent segments. The output is a tuple of
+`SimpleWeightedGraph` and a Dict(label=>vertex) with weights assigned according to `weight_fn`.
+
+```julia
+julia> using ImageSegmentation, Distances
+julia> weight_fn(i,j) = euclidean(seg.segment_pixel_count[i], seg.segment_pixel_count[j]);
```

Comment:
In this section, it might be worth justifying why/whether the pixel count is an interesting/useful means for making decisions about merging. It's also implicit here that `seg` exists, but where did it come from?